api/provider/provider.go

```diff
@@ -43,9 +44,9 @@ type Provider interface {
 
 	LogStream(app string, w io.Writer, opts structs.LogStreamOptions) error
 
-	ReleaseDelete(app, id string) (*structs.Release, error)
+	ReleaseBatchDelete(app, buildID string) error
```

Comment:
We could, but the reason I added it was to keep the the logic for deleting builds and releases separate. That is two calls have to be made for deleting builds and it's releases.

An alternate route I though of is to pass a bool flag to `BuildDelete()` to indicate if the releases should be deleted as well.